### PR TITLE
Exercise-1

### PR DESCRIPTION
**Pull request by Natalia-Melo**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/app.py [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 def index():
-    return 'Hello World!'
\ No newline at end of file
+    items = get_items()
+    return render_template('index.html', items = items)
+
+@app.route('/add_new_item', methods = ['POST'])
+def add_new_item():
+    add_item(request.form.get('title'))
+    return redirect(url_for('index'))
```
Comment (CharlieCumber1) [inline todo_app/app.py:17]: Well done for using `url_for`

File: todo_app/data/session_items.py [hunk 2]
```diff
@@ -38,9 +47,10 @@ def add_item(title):
     items = get_items()
 
     # Determine the ID for the item based on that of the previously added item
-    id = items[-1]['id'] + 1 if items else 0
+    #id = items[-1]['id'] + 1 if items else 0
+
 
-    item = { 'id': id, 'title': title, 'status': 'Not Started' }
+    item = { 'id': len(items), 'title': title, 'status': 'Not Started' }
```
Comment (CharlieCumber1) [inline todo_app/data/session_items.py:53]: Why have you made this change?

I would not have expected any changes in the `sessions_items.py` file.

File: todo_app/templates/index.html [hunk 3]
```diff
@@ -12,8 +11,24 @@ <h1 class="display-4">To-Do App</h1>
...
       <ul class="list-group mb-4">
-        <!-- List the todo items here -->
+        {% for item in items %}
+        <li> {{item.title}} </li>
+        {% endfor %}
       </ul>
     </div>
   </div>
+
+<form action="{{url_for('add_new_item')}}" method="post" class="form-add-item">
```
Comment (CharlieCumber1) [inline todo_app/templates/index.html:21]: Well done for using url_for in the view too.

File: todo_app/templates/index.html [hunk 4]
```diff
@@ -12,8 +11,24 @@ <h1 class="display-4">To-Do App</h1>
...
+  <div class="form-add-item">
+    <label for="name">Name of new item:  </label>
+    <input type="text" name="title" id="name" required>
+  </div>
+  <div class="form-add-item">
+    <button type="submit"> Add new item </button>
+  </div>
+
+
```
Comment (CharlieCumber1) [inline todo_app/templates/index.html:30]: There are quite a few extra new lines here.

**Review by CharlieCumber1 — COMMENTED** [review]
Well Done - you are most of the way there. Let me know if you need any additional information on the changes I have requested.

File: todo_app/data/session_items.py (hunk 2 above)
Comment (Natalia-Melo) [reply todo_app/data/session_items.py]: @CharlieCumber1 I have modified the file back to the original setting.

**Comment by CharlieCumber1** [comment]
Happy for you to merge this when you are ready.

**Review by CharlieCumber1 — APPROVED** [review]
I'm happy for you to merge

**Pull request merged**